docs/latest/user/udp-routing.md

```diff
@@ -0,0 +1,196 @@
+    app: coredns
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: coredns
+  labels:
+    app: coredns
+spec:
+  selector:
+    matchLabels:
+      app: coredns
+  template:
+    metadata:
+      labels:
+        app: coredns
+    spec:
+      containers:
+        - args:
+            - -conf
+            - /root/Corefile
+          image: coredns/coredns
+          name: coredns
+          volumeMounts:
+            - mountPath: /root
+              name: conf
+      volumes:
+        - configMap:
+            defaultMode: 420
+            name: coredns
+          name: conf
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: coredns
+data:
+  Corefile: |
+    .:53 {
+        forward . 8.8.8.8 9.9.9.9
+        log
+        errors
+    }
+
+    foo.bar.com:53 {
+      whoami
+    }
+EOF
+```
+
+Update the Gateway from the Quickstart guide to include a UDP listener that listens on the UDP port `5300`:
+
+```shell
+kubectl patch gateway eg --type=json --patch '[{
+   "op": "add",
+   "path": "/spec/listeners/-",
+   "value": {
+      "name": "coredns",
+      "protocol": "UDP",
+      "port": 5300,
+      "allowedRoutes": {
+         "kinds": [{
+            "kind": "UDPRoute"
+          }]
+      }
+    },
+}]'
+```
+
+Verify the Gateway status:
+
+```shell
+kubectl get gateway/eg -o yaml
+```
+
+## UDPRoute
+
+Create a UDPRoute resource to route UDP traffic received on Gateway port 5300 to the CoredDNS backend.
+
+```shell
+cat <<EOF | kubectl apply -f -
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: UDPRoute
+metadata:
+  name: coredns
+spec:
+  parentRefs:
+    - name: eg
+      sectionName: coredns
+  rules:
+    - backendRefs:
+        - name: coredns
+          port: 53
+EOF
+```
+
+Verify the UDPRoute status:
+
+```shell
+kubectl get udproute/coredns -o yaml
+```
+
+## Testing
+
+Get the External IP of the Gateway:
+
+```shell
+export GATEWAY_HOST=$(kubectl get gateway/eg -o jsonpath='{.status.addresses[0].value}')
+```
+
+Use `dig` command to query the dns entry foo.bar.com through the Gateway.
+
+```shell
+dig @${GATEWAY_HOST} -p 5300 foo.bar.com
+```
+
+You should see the result of the dns query as the below output, which means that the dns query has been successfully
+routed to the backend CoreDNS.
+
+```bash
+; <<>> DiG 9.18.1-1ubuntu1.1-Ubuntu <<>> @10.96.152.156 -p 5300 foo.bar.com
+; (1 server found)
+;; global options: +cmd
+;; Got answer:
+;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 58125
+;; flags: qr aa rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 3
+;; WARNING: recursion requested but not available
+
+;; OPT PSEUDOSECTION:
+; EDNS: version: 0, flags:; udp: 1232
+; COOKIE: 24fb86eba96ebf62 (echoed)
+;; QUESTION SECTION:
+;foo.bar.com.			IN	A
+
+;; ADDITIONAL SECTION:
+foo.bar.com.		0	IN	A	10.244.0.19
+_udp.foo.bar.com.	0	IN	SRV	0 0 42376 .
+
+;; Query time: 1 msec
+;; SERVER: 10.96.152.156#5300(10.96.152.156) (UDP)
+;; WHEN: Fri Jan 13 10:20:34 UTC 2023
+;; MSG SIZE  rcvd: 114
+```
+
```

Comment:
Can you add a __Note:__ `10.96.152.156` is the resolved address of GATEWAY_HOST.